artifacts/package_docker.go

```diff
@@ -78,9 +79,13 @@ func (d *Docker) Builder(ctx context.Context, opts *pipeline.ArtifactContainerOp
 }
 
 func (d *Docker) BuildFile(ctx context.Context, builder *dagger.Container, opts *pipeline.ArtifactContainerOpts) (*dagger.File, error) {
+	// Unlike most other things we push to, docker image tags do not support all characters.
+	// Specifically, the `+` character used in the `buildmetadata` section of semver.
+	version := strings.ReplaceAll(d.Version, "+", "-")
```

Comment:
Is this going to bite us anywhere we need to do any other string matching later down the pipeline?

Reply:
no because this is only when tagging the docker image. The file name and the technical internal version are stored outside of this